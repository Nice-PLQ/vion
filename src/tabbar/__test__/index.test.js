import { mount } from '@vue/test-utils';
import VTabbar from '../tabbar.vue';
import VTabbarItem from '../../tabbar-item';
import VIcon from '../../icon';

describe('Tabbar', () => {
  const fn = jest.fn();
  const wrapper = mount({
    template: `
      <v-tabbar v-model="current" active-color="#00CAFC" inactive-color="#03081A" @change="onChange">
        <v-tabbar-item v-for="(tab, index) in tabs" :key="index">
          <template #icon>
            <v-icon :name="tab.icon" />
          </template>
          {{tab.label}}
        </v-tabbar-item>
      </v-tabbar>
    `,
    data() {
      return {
        current: 0,
        tabs: [
          { label: 'QQ', icon: 'qq' },
          { label: 'QZone', icon: 'qzone' },
          { label: 'Wechat', icon: 'wechat' },
        ]
      };
    },
    methods: {
      onChange: fn
    }
  }, {
    global: {
      components: { VTabbar, VTabbarItem, VIcon }
    }
  });

  test('渲染', async () => {
    const tabItem = wrapper.findAll('.v-tabbar-item')[1];
    await tabItem.trigger('click');
    expect(tabItem.classes()).toContain('on');
    const tabItemContent = tabItem.find('.v-tabbar-item__content');
    expect(tabItemContent.element.style.color).toBe('rgb(0, 202, 252)');
    expect(wrapper.element).toMatchSnapshot();
    expect(fn).toHaveBeenCalled();
  });
});
