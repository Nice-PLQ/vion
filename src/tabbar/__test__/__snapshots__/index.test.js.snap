// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabbar 渲染 1`] = `
<div
  class="v-tabbar"
>
  <ul
    class="v-tabbar__list"
  >
    
    <li
      class="v-tabbar-item"
    >
      <div
        class="v-tabbar-item__content"
        style="color: rgb(3, 8, 26);"
      >
        <div
          class="v-tabbar-item__icon"
        >
          
          <!---->
          
          <!--v-if-->
        </div>
        <div
          class="v-tabbar-item__text"
        >
          
           QQ
          
        </div>
      </div>
    </li>
    <li
      class="v-tabbar-item on"
    >
      <div
        class="v-tabbar-item__content"
        style="color: rgb(0, 202, 252);"
      >
        <div
          class="v-tabbar-item__icon"
        >
          
          <!---->
          
          <!--v-if-->
        </div>
        <div
          class="v-tabbar-item__text"
        >
          
           QZone
          
        </div>
      </div>
    </li>
    <li
      class="v-tabbar-item"
    >
      <div
        class="v-tabbar-item__content"
        style="color: rgb(3, 8, 26);"
      >
        <div
          class="v-tabbar-item__icon"
        >
          
          <!---->
          
          <!--v-if-->
        </div>
        <div
          class="v-tabbar-item__text"
        >
          
           Wechat
          
        </div>
      </div>
    </li>
    
  </ul>
</div>
`;
